### ISSUE TITLE
Gibberish string after deleting last entry in history conatining long strings

### ISSUE THREAD

**Issue by livibetter** [dvorka/hstr#119]
Reproducing steps:

```bash
export HISTFILE=$(mktemp)
: very long string 119 or more, 121 in total, you can press [Alt+1]19x to produce 119 of x's
# invoke hh
# delete the last entry, which should be the `export` command
# press the down key
```

With the steps above, the deleted entry would become a string read like `^X ^P~\^?`, it doesn't seem to matter what the last entry is, as long as there is a long string (121+ chars) in history, it would happen when you delete the last one. But it only would occur for the last one, if you delete any other than the last one, it doesn't happen.

The entry is indeed deleted, just the gibberish string(s) somehow would show up, I had seen two lines of gibberish.

I am not entirely should the amount of characters, but 200 definitely would do.

**Comment by dvorka** [comment]
I did a change that might be root cause of this problem yesterday. Could you please share an example of history file (is I18n?)? Also, does problematic long line contains basic English characters or any language specific characters like Czech (ž, š, á, ...), Chinese or Russian? Thanks!

**Comment by livibetter** [comment]
@dvorka it doesn't matter what the history filename is, the step above uses `/tmp/tmp.something`, it also happens with `.bash_history` that I observed the bug first with. The long string, as my steps described, is a long string with all "x" characters plus a colon and a space, therefore all ASCII letters already can cause it. It happens with both becaeee64ceee8aba285d4ba0085fa341eebed38 and a2b897a8918a58d16e9b2bd629a473e945674022, so it's not what you did yesterday, probably an bug has been around, but no one had noticed.

**Comment by dvorka** [comment]
Unfortunately still not reproducible :-( with ~/.bash_history below:
````
ls -al
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx; 
yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy; 
z120+zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz1; 
```
My method:
* I created the .bash_history with 120+ characters lines ending with `; ` 
* I deleted `z`, `y` and finally `x`

... no problem identified. Where exactly do you see gibberish characters? In the list of history items (e.g. on line that has been deleted)? In the `hh` prompt? After you choose a command?

**Comment by livibetter** [comment]
Shoot, I missed a step, after the deletion, you press down key. The gibberish is in the hh, the same position of the deleted entry, not the history file. Now I knew why you were asking about the filename of history file.

**Comment by dvorka** [comment]
This is helpful! Didn't succeed to reproduce, but will do more experiments later today. Can you please share:
```
hh --version
```
Thanks!

**Comment by livibetter** [comment]
With becaeee64ceee8aba285d4ba0085fa341eebed38:

```
% hh --version
hh version "1.15"
   build   "Jan 14 2015 14:41:19"
```

**Comment by livibetter** [comment]
@dvorka I just tried to use [the entries][], if I inputed in the same order, z, y, x entries were *never* the **last entry** to be deleted, please read and follow my reproducing steps carefully. One of the key point is the "last entry."

Using your entries and deleted the last entry, that gibberish string did show up.

In case you miss again: **delete the last entry**

[the entries]: https://github.com/dvorka/hstr/issues/119#issuecomment-69876607

**Comment by dvorka** [comment]
... unfortunately still not reproducible, however, I'm not going to close this one + I will try to do code review as I will be fixing #123.

**Comment by livibetter** [comment]
Some time after my last comment, I realized that it is actually the *last entry deletion," more like #129, but I didn't bother to report back or change the title of this issue.

I tried to resolve which seemed to work, but I didn't keep the code. From what I could remember, the problem lies around cursor positions and selection size, once you delete the last one, which might be pointing outside of the range. And the `hstr->selectionSize` or cursor positions wasn't updated with `deleteOccurences` if I recall correctly, it's bound to print out something ain't there already.

I did see the correct result, however, I seem to remember during the process, I saw something else causing issues, therefore I didn't keep the code and update this issue.

I think you should deal with #115 first, because you can observe the deletion process more, once it's fixed, then proceed with this and #129.

**Comment by dvorka** [comment]
Fixed along with #129.
